### PR TITLE
[android] Update gradle and other dependencies

### PR DESCRIPTION
**Pull request by biodranik**

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -377,21 +377,21 @@ dependencies {
   // We don't use Kotlin, but some dependencies are actively using it.
   // See https://stackoverflow.com/a/75719642
   implementation 'androidx.core:core:1.13.1'
-  implementation(platform('org.jetbrains.kotlin:kotlin-bom:1.9.23'))
+  implementation(platform('org.jetbrains.kotlin:kotlin-bom:1.9.24'))
```
Comment (Jean-BaptisteC) [inline android/app/build.gradle:380]:
```suggestion
  implementation(platform('org.jetbrains.kotlin:kotlin-bom:2.0.0'))
```

File: android/app/build.gradle [hunk 2]
```diff
@@ -22,18 +22,18 @@ buildscript {
       googleFirebaseServicesDefault
 
   dependencies {
-    classpath 'com.android.tools.build:gradle:8.4.1'
+    classpath 'com.android.tools.build:gradle:8.5.0'
```
Comment (pastk) [inline android/app/build.gradle:25]: What's new?

If there is nothing specific we need asap, then I'd suggest to wait for a couple of patch releases and then update to `8.5.x`

Also upgrading AGP means all android devs will have to upgrade their AS.

**Comment by pastk** [comment]
What's the point to always update to the "latest & greatest" & potentially buggy?

Unless we want some specific new features asap or some known bugfix for a bug that is bugging us.

Otherwise I think its better to play safe and stay on last correction releases.

E.g. now update to 8.4.2.
When 8.6.0 is out update to 8.5.x (whatever number it'll be by that time).
etc.

WDYT?

**Comment by biodranik** [comment]
@vng @rtsisyk @pastk this change has been tested in the latest alpha. It can be safely merged.

File: android/app/build.gradle (hunk 2 above)
Comment (biodranik) [reply android/app/build.gradle]: 8.5.1 is available. Updated the PR.

**Comment by biodranik** [comment]
@vng @rtsisyk PTAL

**Comment by Jean-BaptisteC** [comment]
Be careful about change in androidx.fragment 1.8.0 that's can broken buttons in fragment

**Comment by biodranik** [comment]
@Jean-BaptisteC what are the issues? Any more info about it?

**Review by vng — APPROVED** [review]

**Pull request merged**